### PR TITLE
ARM64: add support for TI AM62X

### PR DESCRIPTION
**Pull request by gramsay0**
This PR adds support for:

1. TI AM62x Cortex-A53 SOC
2. PHYTEC phyCORE-AM62x board

### CODE REVIEW

File: boards/arm64/phycore_am62x_a53/doc/index.rst
Comment (DhruvaG2000) [inline boards/arm64/phycore_am62x_a53/doc/index.rst:102]: We can also probably add a build instruction somewhere,

``west build -p -b phycore_am62x_a53 samples/hello_world``

**Comment by jadonk** [comment]
This is awesome. Do you have any plans to remove the dependency on u-boot?

**Comment by DhruvaG2000** [comment]
image

Attached screenshot of Zephyr SHELL Example up on TI AM625 SK EVM as well (running on A53 core).
Great Work @gramsay0 !

File: soc/arm64/ti_sitara/am6x/mmu_regions.c
Comment (SgrrZhf) [inline soc/arm64/ti_sitara/am6x/mmu_regions.c:26]: Do we need to add `UART` items into `mmu_regions[]`?
The `uart_ns16550` driver can map the mmio address by itself.
https://github.com/zephyrproject-rtos/zephyr/blob/63850804f167353df27b6b33d05f8f3ab642aae4/drivers/serial/uart_ns16550.c#L350

Comment (gramsay0) [reply soc/arm64/ti_sitara/am6x/mmu_regions.c:26]: I wasn't aware of that, I will remove it (this actually explains a lot...)

File: boards/arm64/phycore_am62x_a53/doc/index.rst
Comment (DhruvaG2000) [inline boards/arm64/phycore_am62x_a53/doc/index.rst:71]: Is this really needed? Can you check with upstream UBoot once? AM62 has fairly good upstream uboot support.

**Comment by gramsay0** [comment]
> This is awesome. Do you have any plans to remove the dependency on u-boot?

Thanks! Not currently

File: boards/arm64/phycore_am62x_a53/doc/index.rst
Comment (gramsay0) [reply boards/arm64/phycore_am62x_a53/doc/index.rst:71]: Only these PHYTEC boards are supported in upstream U-Boot:

- phycore-am335x-r2-regor_defconfig
- phycore-am335x-r2-wega_defconfig
- phycore-imx8mm_defconfig
- phycore-imx8mp_defconfig
- phycore-rk3288_defconfig
- phycore_pcl063_defconfig
- phycore_pcl063_ull_defconfig 

But feel free to add TI AM62X EVM boards after this is merged

**Review by carlocaione — DISMISSED** [review]

**Review by povergoing — DISMISSED** [review]

**Review by DhruvaG2000 — DISMISSED** [review]

**Comment by carlocaione** [comment]
> Do you want to add yourself under collaborators for AM62 under https://github.com/zephyrproject-rtos/zephyr/blob/main/MAINTAINERS.yml#L2053 ?

I think we should be a bit more careful with adding as maintainer a first time contributor. No objection as collaborator though.

**Comment by stephanosio** [comment]
> Do you want to ad yourself as maintainer for AM62 under https://github.com/zephyrproject-rtos/zephyr/blob/main/MAINTAINERS.yml#L2053 ?

At this point, the "TI Platforms" area should be split into two different areas: "TI SimpleLink Platforms" and "TI Sitara Platforms," since they are two completely different platforms.

**Review by vaishnavachath — DISMISSED** [review]
@gramsay0 Great work! :confetti_ball:

File: boards/arm64/phycore_am62x_a53/doc/index.rst
Comment (dnltz) [reply boards/arm64/phycore_am62x_a53/doc/index.rst:71]: @gramsay0 - Thanks for this PR! We will send our u-boot upstream soon because we internally focus on the kernel first.

I will send a patch and add the cache cmd to our u-boot.

Comment (gramsay0) [reply boards/arm64/phycore_am62x_a53/doc/index.rst:71]: @dnltz thanks! We can update the docs once that's merged.
`CONFIG_CMD_MP` would also be useful for testing SMP

**Comment by gramsay0** [comment]
I'm not sure why changing MAINTAINERS.yml would break this test:

> tests/lib/mpsc_pbuf/lib.mpsc_pbuf_concurrent (qemu_x86_64:tests/lib/mpsc_pbuf/lib.mpsc_pbuf_concurrent) failed

**Review by SgrrZhf — DISMISSED** [review]

**Review by carlocaione — DISMISSED** [review]

**Comment by carlescufi** [comment]
@gramsay0 can you please rebase this?

File: boards/arm64/phycore_am62x_a53/doc/index.rst
Comment (dnltz) [inline boards/arm64/phycore_am62x_a53/doc/index.rst:9]:
```suggestion
PHYTEC phyCORE-AM62x board is based on TI Sitara applications
```

File: boards/arm64/phycore_am62x_a53/doc/index.rst
Comment (dnltz) [inline boards/arm64/phycore_am62x_a53/doc/index.rst:103]: I'm not a native speaker but feels like there is an article missing.
```suggestion
You can build an application in the usual way. Refer to
```

**Review by dnltz — COMMENTED** [review]
I only found two issues in the documentation. Everything else looks good to me. Thanks!

**Review by dnltz — DISMISSED** [review]

**Review by carlocaione — DISMISSED** [review]

**Review by povergoing — DISMISSED** [review]

**Comment by carlescufi** [comment]
@gramsay0 please note that Zephyr is currently in feature freeze until 3.3.0 is released. See schedule here:
https://github.com/zephyrproject-rtos/zephyr/wiki/Release-Management

**Review by SgrrZhf — DISMISSED** [review]

File: MAINTAINERS.yml
Comment (vaishnavachath) [inline MAINTAINERS.yml:2850]: @gramsay0 nitpick, I think you should rebase to not change the order of maintainers here, see https://github.com/zephyrproject-rtos/zephyr/commit/f05229c6bc7fed9cdbf670f9e16a94e98f945543

**Review by vaishnavachath — CHANGES_REQUESTED** [review]

File: MAINTAINERS.yml
Comment (stephanosio) [inline MAINTAINERS.yml:2849]: Spurious change

**Review by stephanosio — DISMISSED** [review]

File: MAINTAINERS.yml
Comment (gramsay0) [reply MAINTAINERS.yml]: Fixed

**Review by carlocaione — DISMISSED** [review]

**Review by povergoing — DISMISSED** [review]

**Review by SgrrZhf — DISMISSED** [review]

**Review by vaishnavachath — DISMISSED** [review]

**Comment by gramsay0** [comment]
Can we merge this now that 3.3 is released?
It looks like some jobs were cancelled during the last build, is it possible to rebuild without rejecting reviews?

**Review by cfriedt — APPROVED** [review]

**Review by carlocaione — APPROVED** [review]

**Pull request merged**
